chain-signatures/node/src/protocol/signature.rs

```diff
@@ -247,16 +246,17 @@ impl SignatureManager {
         public_key: PublicKey,
         epoch: u64,
         sign_rx: Arc<RwLock<mpsc::Receiver<IndexedSignRequest>>>,
+        msg: &MessageChannel,
```

Comment:
nit: Generally speaking, it's better to take ownership of the parameter when it's needed inside, pushing the clone to the caller side. This may allow the caller to avoid a clone.

Reply:
I think I've heard both sides of this. I guess it's more about what is more expensive to clone. Since `MessageChannel` is more easily cloneable we should have a move instead, but otherwise it should be a reference if it's just read from like a byte vector that can be gigantic if cloned.